src/main/java/bio/terra/app/model/CloudRegion.java

```diff
@@ -0,0 +1,62 @@
+package bio.terra.app.model;
+
+
+import bio.terra.common.EnumUtils;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import com.fasterxml.jackson.databind.exc.InvalidFormatException;
+
+import java.io.IOException;
+
+/**
+ * Represents regions across different cloud providers
+ */
+@JsonDeserialize(using = CloudRegion.CloudRegionDeserializer.class)
+public interface CloudRegion {
+    String name();
```

Comment:
Might be nice to have a javadoc here to indicate what `name()` represents / is used for.

Reply:
yeah, that's a good point. It's admittedly weird since it's an Enum implementing it and that's a built in method